README.md

```diff
@@ -1,63 +1,80 @@
-# To run locally:
+# Conversational Toolkit
 
+Conversational Toolkit is a web application that provides a prototyping and evaluation framework for multimodal voice-driven interactions with conversational AI agents.
 
-## Start backend server:
-To run the server locally, you must first install the Google Cloud CLI and set up your default project and application default credentials.
+The toolkit fronts a language model with automatic speech recognition (ASR) using the WebSpeech API, and text-to-speech (TTS) using the Vertex speech API. It is designed to be modular so that new use cases can be quickly added with well-encapsulated code changes.
 
+## Local Development
 
-Follow [Google Cloud Installation Instructions](https://cloud.google.com/sdk/docs/install) to install the Google Cloud CLI.
+### Start backend server:
+To run the server locally, you must first install the Google Cloud CLI and set up your default project and application default credentials.
 
+Follow [Google Cloud Installation Instructions](https://cloud.google.com/sdk/docs/install) to install the Google Cloud CLI.
 
 Follow [Set up Application Default Credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc) to set up application default credentials.
 
-
 ```sh
 gcloud auth login
 
-
 # Ensure your project has the Vertex APIs enabled
 gcloud config set project [PROJECT_ID]
 gcloud auth application-default login
 gcloud auth application-default set-quota-project [PROJECT_ID]
 ```
 
-
 ```sh
  cd py-server
  pip install -r requirements.txt
  python main.py
 ```
 
-
 You can test the server by pointing your browser at:
 http://localhost:8080/chat?q=hello
 
-
-## Start the client development server:
+### Start the client development server:
 Install npm at: https://nodejs.org/en/download/.
-Install the Angular CLI if necessary: ```npm install -g @angular/cli```
-
 
 Then:
 ```sh
 cd conversational-client
 npm install
-ng serve
+npm run start
 ```
 
-
 Navigate to http://localhost:4200 and click the microphone icon to begin speaking.
 
+## Agent development
+
+Conversational agents are the abstraction used to mediate user/AI interactions.
+An agent is implemented as:
+- A python class on the server that communicates with an LLM (e.g. prompt-chain + supplementary RAG, integrations, world state).
+- An Angular component on the client to drive user interactions through customizable UI.
 
-# To push to Cloud
-Set your cloud project and authenticate as above. Then:
+To create a new agent you will need to add server and client implementations as follows.
 
+#### Server
+- Define a new python file or module that subclasses the `ConversationalAgent` base class and implement the `chat` method.
+  - See agents.py for the class definition, and default_agent.py for an example implementation.
+- Add the new agent with a unique id to `AGENT_BY_ID` in main.py.
```

Comment:
I think we should pull this out of main.py so new agents can be added without changing this core file. Even with agents added internally it'll be easier to review if we can eyeball a review and see that the change doesn't touch any core files.